### PR TITLE
express error handling

### PR DESCRIPTION
**Pull request by rosado**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Bug Fix

## Description

Due to the combination of the following: 
- data fetches that don't go through our generic middleware (`fetchOne`/`fetchMany`) don't handle promise rejections
- express 4.x doesn't handle promise rejections in the middleware

We sometimes end up with unhandled errors (e.g. in `fetchEntry()`).

We can't switch to 5.x (at least not easily, the wizard library relies on something that broke between 4.x and 5.x), so we have to handle this explicitly. This PR adds a wrapper middleware to handle such errors for the few cases where `fetchOne()`/`fetchMany()` isn't used. 500 error is returned now.

Also, moves some of the data fetches to use generic middleware and removes some dead code.

## Related Tickets & Documents

No ticket, but related [Sentry](https://planning-data-platform.sentry.io/issues/5995943040/?environment=production&project=4507033205800960&query=is%3Aunresolved%20issue.priority%3A%5Bhigh%2C%20medium%5D&referrer=issue-stream&statsPeriod=24h&stream_index=0) error

## Added/updated tests?

- [x] Yes

## Release Notes

- **New Features**
	- Introduced a new utility for improved error handling in asynchronous middleware.
	- Added pagination support for issue queries with a defined limit.

- **Bug Fixes**
	- Enhanced error handling to prevent unhandled promise rejections in middleware functions.

- **Documentation**
	- Updated comments and JSDoc annotations to reflect new method signatures and functionalities.

- **Tests**
	- Expanded test coverage for new error handling scenarios and updated existing tests to accommodate changes in data structures.

### CODE REVIEW

File: test/unit/middleware/issueDetails.middleware.test.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ describe('issueDetails.middleware.js', () => {
         // middleware supplies the below
         entryNumber: 1,
         entityCount: { entity_count: 1 },
-        issueEntitiesCount: 1,
+        issueEntitiesCount: { count: 1 },
```
[the remark this answers is not in the document]
Comment (rosado) [reply test/unit/middleware/issueDetails.middleware.test.js:43]: Template params are verified with a schema, this is OK.

File: src/services/performanceDbApi.js [hunk 2]
```diff
@@ -319,66 +286,44 @@ export default {
           return { resource, dataset }
         })
     })
-    return (await Promise.allSettled(requests)).map(p => p.value)
+
+    const results = await Promise.allSettled(requests)
+    return results
+      .filter(p => p.status === 'fulfilled')
+      .map(p => p.value)
```
[the remark this answers is not in the document]
Comment (rosado) [reply src/services/performanceDbApi.js:293]: errors are logged via `.catch(...)` above

[the remark this answers is not in the document]
Comment (rosado) [reply src/services/performanceDbApi.js:293]: already handled

**Review by GeorgeGoodall-GovUk — APPROVED** [review]
looks good!

**Comment by neilfwar** [comment]
showing as done - but no end date?

**Pull request merged**